src/test/resources/case/algorithms/curve25519/main.txt

```diff
@@ -0,0 +1,355 @@
+                if bit <= 18 {
+                    x_1.set(word, x_1.get(word) | { System.shl(value, bit) });
+                } else {
+                    x_1.set(word, x_1.get(word) | { System.shl(value, bit) });
+                    x_1.set(word, x_1.get(word) & 67108863);
+                    x_1.set(word + 1, x_1.get(word + 1) | { System.shr(value, 26 - bit) });
+                };
+                i = i + 1;
+            };
+
+            this.reduceQuick(x_1);
+            this.reduceQuick(x_1);
+        } else {
+            x_1.set(0, 9);
+        };
+        x_2.set(0, 1);
+        i = 0;
+
+        while i < 10 {
+            x_3.set(i, x_1.get(i));
+            i = i + 1;
+        };
+
+        z_3.set(0, 1);
+        var sposn: i32 = 31;
+        var sbit: i32 = 6;
+        var svalue: i32 = System.b2i(privateKey.get(sposn)) | 64;
+        var swap: i32 = 0;
+        var goOn: bool = true;
+        while goOn {
+            var select: i32 = System.shr(svalue, sbit) & 1;
+            swap = swap ^ select;
+            this.swap(swap, x_2, x_3);
+            this.swap(swap, z_2, z_3);
+            swap = select;
+            this.add(A, x_2, z_2);
+            this.square(AA, A);
+            this.sub(B, x_2, z_2);
+            this.square(BB, B);
+            this.sub(E, AA, BB);
+            this.add(C, x_3, z_3);
+            this.sub(D, x_3, z_3);
+            this.multiply(DA, D, A);
+            this.multiply(CB, C, B);
+            this.add(x_3, DA, CB);
+            this.square(x_3, x_3);
+            this.sub(z_3, DA, CB);
+            this.square(z_3, z_3);
+            this.multiply(z_3, z_3, x_1);
+            this.multiply(x_2, AA, BB);
+            this.mulA24(z_2, E);
+            this.add(z_2, z_2, AA);
+            this.multiply(z_2, z_2, E);
+            if sbit > 0 {
+                sbit = sbit - 1;
+            } else if sposn == 0 {
+                goOn = false;
+            } else if sposn == 1 {
+                sposn = 0;
+                svalue = System.b2i(privateKey.get(0)) & 248;
+                sbit = 7;
+            } else {
+                sposn = sposn - 1;
+                svalue = System.b2i(privateKey.get(sposn));
+                sbit = 7;
+            };;;
+        };
+        this.swap(swap, x_2, x_3);
+        this.swap(swap, z_2, z_3);
+        this.modInv(z_3, z_2);
+        this.multiply(x_2, x_2, z_3);
+        {
+            var index: i32 = 0;
+
+            while index < 32 {
+                var bit: i32 = index * 8 % 26;
+                var word: i32 = index * 8 / 26;
+                if bit <= 18 {
+                    result.set(index, System.i2b(System.shr(x_2.get(word), bit)));
+                } else {
+                    result.set(index, System.i2b(System.shr(x_2.get(word), bit) | System.shl(x_2.get(word + 1), 26 - bit)));
+                };
+                index = index + 1;
+            };
+        }
+    }
+
+    fn new() super()
+}
```

Comment:
I would've preferred moving this to `curve25519.txt` or `lib.txt` but idk how you import stuff